metpy/calc/kinematics.py

```diff
@@ -658,31 +659,50 @@ def potential_vorticity_baroclinic(potential_temperature, pressure, u, v, dx, dy
 
     Returns
     -------
-    (M, N) ndarray
+    (P, M, N) ndarray
         baroclinic potential vorticity
 
     Notes
     -----
-    The same formula is used for isobaric and isentropic PV analysis. Provide winds
-    for vorticity calculations on the desired isobaric or isentropic surface. Three layers
-    of pressure/potential temperature are required in order to calculate the vertical
-    derivative (one above and below the desired surface).
+    This function will only work with data that is in (P, Y, X) format. If your data
+    is in a different order you will need to re-order your data in order to get correct
+    results from this function.
+
+    The same function can be used for isobaric and isentropic PV analysis. Provide winds
+    for vorticity calculations on the desired isobaric or isentropic surface. At least three
+    layers of pressure/potential temperature are required in order to calculate the vertical
+    derivative (one above and below the desired surface). The first two terms will be zero if
+    isentropic level data is used due to the gradient of theta in both the x and y-directions
+    will be zero since you are on an isentropic surface.
+
+    This function expects pressure/isentropic level to increase with increasing array element
+    (e.g., from higher in the atmosphere to closer to the surface. If the pressure array is
+    one-dimensional p[:, None, None] can be used to make it appear multi-dimensional.)
 
     """
-    if np.shape(potential_temperature)[axis] != 3:
-        raise ValueError('Length of potential temperature along axis '
-                         '{} must be 3.'.format(axis))
-    if np.shape(pressure)[axis] != 3:
+    if np.shape(potential_temperature)[-3] < 3:
+        raise ValueError('Length of potential temperature along the pressure axis '
+                         '{} must be at least 3.'.format(-3))
+    if np.shape(pressure)[-3] < 3:
         raise ValueError('Length of pressure along axis '
-                         '{} must be 3.'.format(axis))
-    avor = absolute_vorticity(u, v, dx, dy, lats, dim_order=dim_order)
-    stability = first_derivative(potential_temperature, x=pressure, axis=axis)
-    # Get the middle layer stability derivative (index 1)
-    slices = [slice(None)] * stability.ndim
-    slices[axis] = 1
+                         '{} must be at least 3.'.format(-3))
+    avor = absolute_vorticity(u, v, dx, dy, lats, dim_order='yx')
+    dthtadp = first_derivative(potential_temperature, x=pressure, axis=-3)
+
+    if np.shape(potential_temperature)[1] == 1:
```

Comment:
I was trying to eliminate these repeated blocks, but don't see a great way that doesn't make the following calculation less obvious.

Reply:
Here, if one (x-dimension) is only length 1, then the other (y-dimension) would also be length 1 by definition. Theta would (should?) not vary in x or y if you are on an isentropic surface, which is what this portion is attempting to cover. Could move to something like...

```python
if (np.shape(potential_temperature)[1] == 1) and (np.shape(potential_temperature[2] == 1):
    dthtady = 0 * units.K / units.m  # axis=1 only has one dimension
    dthtadx = 0 * units.K / units.m  # axis=2 only has one dimension
else:
    dthtady = first_derivative(potential_temperature, delta=dy, axis=-2)
    dthtadx = first_derivative(potential_temperature, delta=dx, axis=-1)
```
Added complexity is minimal?

Reply:
This seems ok to me.